src/huggingface_hub/inference/_common.py

```diff
@@ -265,34 +265,44 @@ def _bytes_to_image(content: bytes) -> "Image":
 ## STREAMING UTILS
 
 
-def _stream_text_generation_response(
+def _stream_text_generation_response_from_bytes(
     bytes_output_as_lines: Iterable[bytes], details: bool
 ) -> Union[Iterable[str], Iterable[TextGenerationStreamOutput]]:
     """Used in `InferenceClient.text_generation`."""
     # Parse ServerSentEvents
     for byte_payload in bytes_output_as_lines:
-        output = _format_text_generation_stream_output(byte_payload, details)
+        try:
+            output = _format_text_generation_stream_output(byte_payload, details)
+        except StopIteration:
+            break
         if output is not None:
             yield output
 
 
-async def _async_stream_text_generation_response(
+async def _async_stream_text_generation_response_from_bytes(
     bytes_output_as_lines: AsyncIterable[bytes], details: bool
 ) -> Union[AsyncIterable[str], AsyncIterable[TextGenerationStreamOutput]]:
     """Used in `AsyncInferenceClient.text_generation`."""
     # Parse ServerSentEvents
     async for byte_payload in bytes_output_as_lines:
-        output = _format_text_generation_stream_output(byte_payload, details)
+        try:
+            output = _format_text_generation_stream_output(byte_payload, details)
+        except StopIteration:
+            break
         if output is not None:
             yield output
 
 
 def _format_text_generation_stream_output(
     byte_payload: bytes, details: bool
 ) -> Optional[Union[str, TextGenerationStreamOutput]]:
+    print(byte_payload)
     if not byte_payload.startswith(b"data:"):
         return None  # empty line
 
+    if byte_payload == b"data: [DONE]":
+        raise StopIteration("[DONE] signal received.")
```

Comment:
Wouldn't yielding None have the same effect, but saving the extra try except ?

Reply:
yielding None would work but the StopIteration ensures no new messages are read from the server (closer to [openai's implementation](https://github.com/openai/openai-python/blob/1ed0e35679438cc7409f2e3aeb36d637b3c579c0/src/openai/_streaming.py#L60) that break on `[DONE]` signal)